src/saveEventsFile.m

```diff
@@ -146,36 +153,65 @@
 end
 
 function logFile = initializeFile(cfg, logFile)
+% This function creates the bids field structure for json files for the
+% three basic bids event columns, and for all requested extra columns. 
+%
+% Note that subfields (e.g. unit, levels etc. can be changed by the user 
+% before calling openFile.  
+
+    % initialize holy trinity (onset, trial_type, duration) columns
+    logFile(1).columns = struct( ...
+                         'onset', struct( ...
+                                         'Description', 'time elapsed since experiment start', ...
+                                         'Unit', 's'), ...
+                         'trial_type', struct( ...
+                                              'Description', 'types of trial', ...
+                                              'Levels', ''), ...
+                         'duration', struct( ...
+                                            'Description', 'duration of the event or the block', ...
+                                            'Unit', 's') ...
+                        );   
 
-    logFile = initializeStimFile(cfg, logFile);
-
-    % print the basic BIDS columns
-    fprintf(logFile(1).fileID, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
-    fprintf(1, '%s\t%s\t%s', 'onset', 'duration', 'trial_type');
 
-    printHeaderExtraColumns(logFile);
-
-    % next line so we start printing at the right place
-    fprintf(logFile(1).fileID, '\n');
-    fprintf(1, '\n');
+    logFile = initializeExtraColumns(logFile);
 
 end
 
 function logFile = initializeStimFile(cfg, logFile)
```

Comment:
I was thinking about the same thing, and in the end I retained initializeStimFile as a separate function. 

I slightly prefer this because it allows for expansion of that function in the future (perhaps at some point we'll want to do more than just initializeExtraColumns?). It also makes it little bit nicer to read the code, as you mentioned.

Reply:
I agree. :wink: